js/src/cli.ts

```diff
@@ -709,18 +713,32 @@ async function collectFiles(
 // In addition to marking node_modules external, explicitly mark
 // our packages (braintrust and autoevals) external, in case they're
 // installed in a relative path.
-const markKnownPackagesExternalPlugin = {
-  name: "make-known-packages-external",
-  setup(build: esbuild.PluginBuild) {
-    // Mark known packages as external
-    const knownPackagesFilter =
-      /^(braintrust|autoevals|@braintrust\/|config|lightningcss)/;
-    build.onResolve({ filter: knownPackagesFilter }, (args) => ({
-      path: args.path,
-      external: true,
-    }));
-  },
-};
+function createMarkKnownPackagesExternalPlugin(
+  additionalPackages: string[] = [],
+) {
+  return {
+    name: "make-known-packages-external",
+    setup(build: esbuild.PluginBuild) {
+      // Mark known packages as external
+      const knownPackages = [
+        "braintrust",
+        "autoevals",
+        "@braintrust/",
+        "config",
+        "lightningcss",
+        "@mapbox/node-pre-gyp",
+        ...additionalPackages,
+      ];
+      const knownPackagesFilter = new RegExp(
+        `^(${knownPackages.map((pkg) => pkg.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")).join("|")})`,
```

Comment:
didn't test but perhaps this so that you don't catch all package names that happen to have these prefixes

```suggestion
        "@braintrust",
        "config",
        "lightningcss",
        "@mapbox/node-pre-gyp",
        ...additionalPackages,
      ];
      const knownPackagesFilter = new RegExp(
        `^(${knownPackages.map((pkg) => pkg.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")).join("|")})(?:/|$)`,
```

Reply:
I made a more invasive change here: https://github.com/braintrustdata/braintrust-sdk/pull/790/files#diff-4f2328e013899dee49744473623d1e46ecec44c3acdf5e10bf908edd59c04413R24